pkg/storage/internal/actions/create_process_inputs.go

```diff
@@ -0,0 +1,273 @@
+		return fmt.Errorf("failed to check if transaction are known: %w", err)
+	}
+
+	if !allKnown {
+		return fmt.Errorf("some tx in the inputBEEF is not known to storage; valid and contain complete proof data for transactions: %s", strings.Join(notProvidedInInputsTxs, ", "))
+	}
+
+	return nil
+}
+
+func (proc *inputsProcessor) checkInputsAndMergeTxIDsToBEEF() error {
+	missingFullProofs := seq.Collect(seq.Filter(maps.Keys(proc.txIDsLookup), func(txID string) bool {
+		btx, ok := proc.beef.Transactions[txID]
+		return !ok || btx.DataFormat == transaction.TxIDOnly
+	}))
+
+	if len(missingFullProofs) == 0 {
+		return nil
+	}
+
+	if !proc.trustSelf {
+		return fmt.Errorf("valid and contain complete proof data for transactions: %s", strings.Join(missingFullProofs, ", "))
+	}
+
+	allKnown, err := proc.parent.provenTxRepo.ExistsAllProvenTxs(proc.ctx, missingFullProofs, readyToBeInputProvenTxStatuses)
+	if err != nil {
+		return fmt.Errorf("failed to check if transactions are known: %w", err)
+	}
+
+	if !allKnown {
+		return fmt.Errorf("some tx used in provided input is not known to storage; valid and contain complete proof data for transactions: %s", strings.Join(missingFullProofs, ", "))
+	}
+
+	for _, txID := range missingFullProofs {
+		proc.beef.MergeTxidOnly(txID)
+	}
+
+	return nil
+}
+
+func (proc *inputsProcessor) xinputDefOnKnownUTXO(xinput *wdk.ValidCreateActionInput, output *wdk.TableOutput) (*xinputDefinition, error) {
+	if output.LockingScript == nil || len(*output.LockingScript) == 0 || output.Satoshis <= 0 {
+		return nil, fmt.Errorf("output %s has no locking script or positive satoshis", xinput.Outpoint)
+	}
+
+	if !output.Spendable {
+		return nil, fmt.Errorf("output %s is not spendable", xinput.Outpoint)
+	}
+
+	return &xinputDefinition{
+		ValidCreateActionInput: xinput,
+		Satoshis:               satoshi.MustFrom(output.Satoshis),
+		LockingScript:          *output.LockingScript,
+		knownOutput:            output,
+	}, nil
+}
+
+func (proc *inputsProcessor) xinputDefOnUnknownUTXO(xinput *wdk.ValidCreateActionInput) (*xinputDefinition, error) {
+	btx, ok := proc.beef.Transactions[xinput.Outpoint.TxID]
+	if !ok || btx == nil {
+		return nil, fmt.Errorf("input %s not found in beef or outputs", xinput.Outpoint)
+	}
+
+	if btx.DataFormat == transaction.TxIDOnly {
+		beefForTx, err := proc.parent.provenTxRepo.BuildValidBEEF(proc.ctx, xinput.Outpoint.TxID, readyToBeInputProvenTxStatuses)
+		if err != nil {
+			return nil, fmt.Errorf("failed to build beef for tx %s: %w", xinput.Outpoint.TxID, err)
+		}
+
+		btx, ok = beefForTx.Transactions[xinput.Outpoint.TxID]
+		if !ok || btx == nil {
+			return nil, fmt.Errorf("tx %s not found in beef", xinput.Outpoint.TxID)
+		}
+
+		if _, err = proc.beef.MergeBeefTx(btx); err != nil {
+			return nil, fmt.Errorf("failed to merge beef for tx %s: %w", xinput.Outpoint.TxID, err)
+		}
+	}
+
+	voutInt := must.ConvertToIntFromUnsigned(xinput.Outpoint.Vout)
+	if voutInt >= len(btx.Transaction.Outputs) {
```

Comment:
```suggestion
	if voutInt < 0 || voutInt >= len(btx.Transaction.Outputs) {
```

think negative case should be impossible, but just in case.

Reply:
This `voutInt` is created one line above and it's created from `unsigned` so there is no such possibility to be negative. 
It would be "over-deffensive" programming 😅